code/_onclick/hud/meter/meter.dm

```diff
@@ -51,74 +50,26 @@
 
 	var/margin = 8//Extra length that isn't counted as part of our length for the purpose of components
 
-/obj/screen/meter/New(var/atom/holder)
-
+/obj/screen/meter/New(atom/holder)
+	L = holder
 	cache_data(arglist(args))
-	.=..()
 
+	.=..()
 
 //Override this and change the parameters in subtypes
-/obj/screen/meter/proc/cache_data(var/atom/holder)
+/obj/screen/meter/proc/cache_data(atom/holder)
 
 
 /obj/screen/meter/Destroy()
 	QDEL_NULL(remaining_meter)
 	QDEL_NULL(delta_meter)
 	QDEL_NULL(limit_meter)
 	QDEL_NULL(textholder)
-	if (C)
```

Comment:
cleaning itself up from the client screen makes sense

Reply:
> cleaning itself up from the client screen makes sense


It's better to never work directly with such important things.